src/app/groups/groups.datasource.ts

```diff
@@ -33,14 +32,32 @@ export class GroupsDataSource implements DataSource<any> {
    * @param {number} limit Number of entries within the page.
    * @param {boolean} groupActive Specify whether to only filter active groups.
    */
-  getGroups(filterBy: any, orderBy: string = '', sortOrder: string = '', pageIndex: number = 0, limit: number = 10, groupActive: boolean = true) {
+  getGroups(
+    filterBy: any,
+    orderBy: string = "",
+    sortOrder: string = "",
+    pageIndex: number = 0,
+    limit: number = 10,
+    groupActive: boolean = true
+  ) {
     this.groupsSubject.next([]);
-    this.groupsService.getGroups(filterBy, 'name', 'ASC', pageIndex, limit)
-      .subscribe((groups: any) => {
-        groups.pageItems = (groupActive) ? (groups.pageItems.filter((group: any) => group.active)) : groups.pageItems;
+    let countryId = JSON.parse(sessionStorage.getItem("selectedCountry"))?.id;
+
+    if (countryId) {
+      this.groupsService
```

Comment:
pleas apply here as well the comment left in client component of grouping the duplicate outside the if else blocks